### PR TITLE
[release-4.15] OCPBUGS-27221: Bump library-go to deal with GET conflict

### PR DESCRIPTION
**Pull request by jhadvig**
Manual backport of https://github.com/openshift/console-operator/pull/836

/assign @stlaz @TheRealJon

### CODE REVIEW

**Comment by jhadvig** [comment]
/hold 
wrong lib-go version

File: go.mod [hunk 1]
```diff
@@ -9,18 +9,18 @@ require (
 	github.com/go-bindata/go-bindata v3.1.2+incompatible
 	github.com/go-test/deep v1.0.5
 	github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822
-	github.com/openshift/api v0.0.0-20231128131348-2eef5be42ab4
+	github.com/openshift/api v0.0.0-20231218131639-7a5aa77cc72d
```
Comment (stlaz) [inline go.mod:12]: this is a ref to master

File: go.mod [hunk 2]
```diff
@@ -9,18 +9,18 @@ require (
...
 	github.com/openshift/build-machinery-go v0.0.0-20220913142420-e25cf57ea46d
-	github.com/openshift/client-go v0.0.0-20230926161409-848405da69e1
-	github.com/openshift/library-go v0.0.0-20230927113136-405c34317fa4
+	github.com/openshift/client-go v0.0.0-20231218140158-47f6d749b9d9
```
Comment (stlaz) [inline go.mod:14]: this is a ref to master

File: go.mod [hunk 3]
```diff
@@ -9,18 +9,18 @@ require (
...
+	github.com/openshift/library-go v0.0.0-20240117151256-95b334bccb5d
 	github.com/spf13/cobra v1.7.0
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.28.2
-	k8s.io/apiextensions-apiserver v0.28.2
-	k8s.io/apimachinery v0.28.2
-	k8s.io/client-go v0.28.2
-	k8s.io/component-base v0.28.2
-	k8s.io/klog/v2 v2.100.1
+	k8s.io/api v0.29.0
```
Comment (stlaz) [inline go.mod:18]: 4.15 is k8s 1.28 IIRC

**Review by stlaz — CHANGES_REQUESTED** [review]

**Comment by jhadvig** [comment]
/retest


Flake
```
--- FAIL: TestLegacyCustomURLWithIngressConsoleComponentRoute (37.10s)
```

**Pull request merged**